packages/app-mobile/components/NoteEditor/EditLinkDialog.tsx

```diff
@@ -0,0 +1,156 @@
+// Dialog allowing the user to update/create links
+
+const React = require('react');
+const { useState, useEffect, useMemo, useRef } = require('react');
+const { StyleSheet } = require('react-native');
+const { View, Modal, Text, TextInput, Button } = require('react-native');
+
+import { themeStyle } from '@joplin/lib/theme';
+import { _ } from '@joplin/lib/locale';
+import { EditorControl } from './types';
+import SelectionFormatting from './SelectionFormatting';
+
+interface LinkDialogProps {
+	editorControl: EditorControl;
+	selectionState: SelectionFormatting;
+	visible: boolean;
+	themeId: number;
+}
+
+const EditLinkDialog = (props: LinkDialogProps) => {
+	// The content of the link selected in the editor (if any)
+	const editorLinkData = props.selectionState.linkData;
+	const [linkLabel, setLinkLabel] = useState('');
+	const [linkURL, setLinkURL] = useState('');
+
+	// Reset the label and URL when shown/hidden
+	useEffect(() => {
+		setLinkLabel(editorLinkData.linkText ?? props.selectionState.selectedText);
+		setLinkURL(editorLinkData.linkURL ?? '');
+	}, [props.visible]);
```

Comment:
Please make sure to add all dependencies to the array.

But it should strictly be what's required and nothing more - so `editorLinkData.linkText, props.selectionState.selectedText, editorLinkData.linkURL` (and **not** `editorLinkData, props.selectionState`)